conda_build/cli/main_index.py

```diff
@@ -85,7 +87,7 @@ def parse_args(args):
     p.add_argument(
         "-f",
         "--file",
-        help="A file that contains a new line separated list of packages to add to repodata.",
+        help="A file that contains a new line separated list of packages to add to repodata.  Deprecated, will be removed in a future version of conda build.",
```

Comment:
I know that conda-build hasn't officially adopted the [Conda Release Schedule](https://github.com/conda-incubator/ceps/blob/main/cep-8.md) but I still think it would be a good idea to provide a version number when this will be removed. Otherwise, it probably never will 🤷.

Reply:
> Thanks Daniel!
> 
> I have some questions:
> 
> * What's the story for users of the `conda index` subcommand in this PR? Are they silently being upgraded to the `conda-index` rewrite? If that's the case, what backwards incompatible changes should they expect?
> * Same but for users of `conda_build.api`.

Yes, everyone is upgraded to the new conda-index with the possible exception of people who import parts of this file that we have left in for backwards compatibility to follow a deprecation policy that this project has not officially adopted.

Users should expect equivalent indexing with a sqlite cache instead of a many-small-files cache.

In this PR we try to map all the API calls to forwarded calls to new-conda-index.